### DISCUSSION TITLE
Combine Prisma schema files with code

### DISCUSSION THREAD

**Discussion by BossDaily** [prisma/prisma#18289] [Q&A]
I'm currently working on a plugin system that requires me to combine multiple Prisma schema files programmatically. The system involves creating and editing models, and I'm finding it quite challenging to manage multiple schema files separately.

My goal is to find a way to combine all the schema files together into a single file that I can use to generate the Prisma client. This is for a plugins system for one of my projects where other developers can store settings or other information in the same database that the main project uses.

I'm wondering if anyone else has faced a similar problem and has found a good solution. Perhaps there's a library or tool out there that can help me achieve my goal? Or if there is an alternative way I should be solving this issue that will be appreciated. 

I appreciate any suggestions or advice you can offer. Thanks in advance!

**Comment by ludralph** [comment]
Hi @BossDaily 👋 

You can have multiple schema files and run them individually, however, you cannot combine schemas into one generated client.
In other words, with Prisma, you cannot split a large schema into smaller, more organized schemas. There is a very active [issue on github](https://github.com/prisma/prisma/issues/2377) tracking the discussion on this topic. 

However, there are third-party libraries that you can leverage to combine multiple schema files. You can take a look at either `prisma-merge` library on npm or `prisma-util` library also on npm.

**Comment by BossDaily** [comment]
For anyone reading this in the future another good library I found is 
https://www.npmjs.com/package/prismix

**Comment by ryanhsmfc** [comment]
just put a bookmark here in case anyone looking for the answer:

prisma has folder feature to put multiple schema. 

https://www.prisma.io/blog/organize-your-prisma-schema-with-multi-file-support

`previewFeatures = ["prismaSchemaFolder"]`

**Comment by Adil512** [comment]
Schemas can be combine easily using online tools like **[https://schemabuilder.net/schema-combiner/](https://schemabuilder.net/schema-combiner/)**
I just tried for testing purposes and also validated with Google Rich Result Checking tool and it is combining correctly
